app/routes/blog.$slug.tsx

```diff
@@ -0,0 +1,38 @@
+import { type LoaderArgs } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import { getMDXComponent } from "mdx-bundler/client";
+import { useMemo } from "react";
+import invariant from "tiny-invariant";
+import { PUBLISH_DATE_FORMATTER } from "~/utils";
+import { type Mdx } from "blog/parser";
```

Comment:
Nit: 

```suggestion
import  type { Mdx } from "blog/parser";
```

I didn't even realize this was possible 😅. It's really not a big deal, just adding the suggestion for consistency's sake.